ostd/src/arch/x86/iommu/fault.rs

```diff
@@ -201,20 +210,61 @@ bitflags! {
     }
 }
 
-pub(super) static FAULT_EVENT_REGS: Once<FaultEventRegisters> = Once::new();
+pub(super) static FAULT_EVENT_REGS: Once<SpinLock<FaultEventRegisters, LocalIrqDisabled>> =
+    Once::new();
 
 /// Initializes the fault reporting function.
 ///
 /// # Safety
 ///
 /// User must ensure the base_register_vaddr is read from DRHD
 pub(super) unsafe fn init(base_register_vaddr: Vaddr) {
-    FAULT_EVENT_REGS.call_once(|| FaultEventRegisters::new(base_register_vaddr));
+    FAULT_EVENT_REGS.call_once(|| SpinLock::new(FaultEventRegisters::new(base_register_vaddr)));
+}
+
+fn iommu_fault_handler(frame: &TrapFrame) {
+    let mut fault_event_regs = FAULT_EVENT_REGS.get().unwrap().lock();
+
+    primary_fault_handler(&mut fault_event_regs);
+
+    let fault_status = fault_event_regs.status();
+    if fault_status.contains(FaultStatus::IQE | FaultStatus::ICE | FaultStatus::ITE) {
```

Comment:
Is this condition correct? This condition implies that all three statuses must be set. However, according to the definition, if any status is set, it should trigger an invalidation error.

Should it be:
```rust
if fault_status.intersects(FaultStatus::IQE | FaultStatus::ICE | FaultStatus::ITE)
```